thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java

```diff
@@ -72,7 +72,7 @@ public TimeOnTimeResponseParser(ThirdEyeResponse baselineResponse, ThirdEyeRespo
 
     metricFunctions = baselineResponse.getMetricFunctions();
     numMetrics = metricFunctions.size();
-    numTimeBuckets = baselineRanges.size();
+    numTimeBuckets = Math.min(currentRanges.size(), baselineRanges.size());
```

Comment:
Add a log when the ranges are not equal and chose the smaller among the two. Explicitly state that this can happen during DST